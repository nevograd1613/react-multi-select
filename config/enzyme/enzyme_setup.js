// TODO: Remove this `raf` polyfill once the below issue is sorted
// https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-332842582
import raf from "./tempPolyfills";

import Enzyme from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";

Enzyme.configure({ adapter: new Adapter() });
