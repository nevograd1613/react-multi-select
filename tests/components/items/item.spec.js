import React from "react";
import ShallowRenderer from "react-shallow-renderer";
import { shallow } from "enzyme";

import Item from "../../../src/components/items/item";

const item = { id: 1, label: "Hi" };

describe("Item", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with valid item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item item={item} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with item without id", () => {
    const itemWithoutId = { label: "Hi" };

    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item item={itemWithoutId} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with checked status", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item checked={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with indeterminate status", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item indeterminate={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with indeterminate and checked status", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item indeterminate={true} checked={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with border", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item withBorder={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with disabled", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item disabled={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with height", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item height={20} />);
    expect(tree).toMatchSnapshot();
  });

  test("click will trigger onClick", () => {
    const onClick = jest.fn();
    const item = shallow(<Item onClick={onClick} />);
    item.simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("snapshot with group item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item item={item} group />);
    expect(tree).toMatchSnapshot();
  });
});
