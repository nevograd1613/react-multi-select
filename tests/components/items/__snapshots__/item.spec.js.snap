// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item default snapshot 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with border 1`] = `
<div
  className="item with_border"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with checked status 1`] = `
<div
  className="item selected"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={true}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with disabled 1`] = `
<div
  className="item disabled"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={true}
    id={null}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with group item 1`] = `
<div
  className="item with_grouping"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <ItemLabel
    label="Hi"
  />
</div>
`;

exports[`Item snapshot with height 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    {
      "height": 20,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with indeterminate and checked status 1`] = `
<div
  className="item selected"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={true}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={true}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with indeterminate status 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={true}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with item without id 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    id={null}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label="Hi"
  />
</div>
`;

exports[`Item snapshot with valid item 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    id="checkbox-1"
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label="Hi"
  />
</div>
`;
